Trying to wrap a C++ class that is non-copyable will fail to compile, unless a operator<< is defined for the class, too.

**Describe the Bug**
When using vrpc to wrap a C++ class into a nodejs addon, this will fail to compile (and work) if the C++ class is "non-copyable", i.e. its copy constructor is deleted. I encountered this while trying to wrap a C++ class that was using some thread and mutex members which are non-copyable.

**Steps To Reproduce**
In the file `examples/Bar.hpp`, add a deleted copy constructor in the class declaration, i.e. a line saying 
`Bar(const Bar&) = delete;`

Then build the project by calling `npm run build`. Resulting compile error is this:

> In file included from ../vrpc/addon.cpp:42:
> ../vrpc/vrpc.hpp: In instantiation of 'vrpc::Value::holder<T>::holder(const T&) [with T = vrpc_example::Bar]':
> ../vrpc/vrpc.hpp:501:15:   required from 'vrpc::Value::Value(const T&) [with T = vrpc_example::Bar]'
> ../vrpc/vrpc.hpp:445:21:   required from 'std::string vrpc::to_string(const T&) [with T = vrpc_example::Bar; std::string = std::__cxx11::basic_string<char>]'
> ../vrpc/vrpc.hpp:691:31:   required from 'std::string vrpc::Value::holder<std::shared_ptr<_Tp> >::format() const [with T = vrpc_example::Bar; std::string = std::__cxx11::basic_string<char>]'
> ../vrpc/vrpc.hpp:690:27:   required from here
> ../vrpc/vrpc.hpp:643:14: error: use of deleted function 'vrpc_example::Bar::Bar(const vrpc_example::Bar&)'
>   643 |       : held(new T(value)) { }
>       |              ^~~~~~~~~~~~
> In file included from ../examples/binding.cpp:2,
>                  from ../vrpc/addon.cpp:45:
> ../examples/Bar.hpp:25:5: note: declared here
>    25 |     Bar(const Bar&) = delete;
>       |     ^~~
> make: *** [vrpc_example.target.mk:109: Release/obj.target/vrpc_example/vrpc/addon.o] Error 1
> make: Leaving directory '/home/cs/org/vrpc/build'

**Expected Behavior**
It is expected to be able to wrap a non-copyale C++ class, too. The compiler instantiation stack seems to indicate that the copy constructor was instantiated "only" for the `to_string` function and this seems unnecessary to me. The stack seems to indicate that the value object was wrapped in a shared_ptr anyway (line vrpc.hpp:691), but then from `to_string` the copy constructor of `Value` is called directly (line vrpc.hpp:445), which on the other hand seems rather weird to me. Maybe you need to do some tricks in `to_string` so that no copy constructor of `Value` is called?

**Version**
Current git master, i.e. version 2.2.3.

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,5 +1,3 @@
-before_install:
-
 matrix:
   include:
     - language: node_js

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 - Wrong `break` statements when inspecting the client cache for instances to be
   attached to [NodeJS]
+- Compilation error on non-copyable classes [C++-Addon] (#22)
 
 ## [2.3.0] - 13 Jan 2021
 

--- a/test/cpp/vrpcTest.cpp
+++ b/test/cpp/vrpcTest.cpp
@@ -60,35 +60,4 @@ TEST_CASE("Test functionality of generic holder", "[Value]") {
     auto ia = v.get<int>();
     REQUIRE(ia == 1);
   }
-  /* NOTE: Currently unsupported because of windows compiler issues
-  SECTION("Check retrieval without get") {
-    // Store an integer to Value
-    Value v(1);
-    // by value
-    int iv = v;
-    REQUIRE(iv == 1);
-    // by const reference
-    const int& ir = v;
-    REQUIRE(ir == 1);
-    // by std::shared_ptr
-    std::shared_ptr<int> ip = v;
-    REQUIRE(*ip == 1);
-  }
-  */
-  SECTION("Check formatting") {
-    Value v;
-    v = 1;
-    REQUIRE(v.format() == "1");
-    v = "foo";
-    REQUIRE(v.format() == "foo");
-    v = 3.1414;
-    REQUIRE(v.format() == "3.141400");
-    v = std::vector<int>({1, 2, 3, 4});
-    REQUIRE(v.format() == "1,2,3,4");
-    v = std::map<std::string, int>({
-      {"1", 1},
-      {"2", 2}});
-    REQUIRE(v.format() == "{1:1,2:2}");
-    v = std::set<int>({1, 2, 3, 4});
-  }
 }

--- a/vrpc/vrpc.hpp
+++ b/vrpc/vrpc.hpp
@@ -439,47 +439,6 @@ namespace vrpc {
     return signature.empty() ? signature : "-" + signature;
   }
 
-  template <typename T>
-  inline std::string to_string(const T& value) {
-    std::ostringstream s;
-    s << std::fixed << value;
-    return s.str();
-  }
-
-  template <typename T>
-  inline std::string to_string(const std::shared_ptr<T>& value) {
-    std::ostringstream s;
-    s << std::fixed << *value;
-    return s.str();
-  }
-
-  template <typename T>
-  inline std::string to_string(const std::vector<T>& value) {
-    if (value.empty()) return "";
-    std::ostringstream s;
-    typename std::vector<T>::const_iterator it = value.begin();
-    s << to_string(*it);
-    it++;
-    for (; it != value.end(); ++it) {
-      s << "," << to_string(*it);
-    }
-    return s.str();
-  }
-
-  template <typename KeyType, typename ValueType>
-  inline std::string to_string(const std::map<KeyType, ValueType>& value) {
-    if (value.empty()) return "{}";
-    std::ostringstream s;
-    typename std::map<KeyType, ValueType>::const_iterator it = value.begin();
-    s << "{" << to_string(it->first) << ":" << to_string(it->second);
-    it++;
-    for (; it != value.end(); ++it) {
-      s << "," << to_string(it->first) << ":" << to_string(it->second);
-    }
-    s << "}";
-    return s.str();
-  }
-
   namespace detail {
 
     template <typename T>
@@ -590,12 +549,6 @@ namespace vrpc {
       return content ? content->type() : void_type;
     }
 
-    std::string format() const {
-      return content ? content->format() : std::string();
-    }
-
-    friend std::ostream& operator<<(std::ostream& os, const Value& v);
-
     template <typename T>
     inline const T& get() const {
       // TODO Check type correctness here
@@ -631,8 +584,6 @@ namespace vrpc {
       virtual std::type_index type() const noexcept = 0;
 
       virtual placeholder* clone() const = 0;
-
-      virtual std::string format() const = 0;
     };
 
     template<typename T>
@@ -655,10 +606,6 @@ namespace vrpc {
       virtual placeholder* clone() const {
         // This copies a std::shared_ptr!
         return new holder(*held);
-      }
-
-      virtual std::string format() const {
-        return vrpc::to_string(*held);
       }
 
       holder& operator=(const holder&) = delete;
@@ -687,12 +634,7 @@ namespace vrpc {
         return new holder(held);
       }
 
-      virtual std::string format() const {
-        return vrpc::to_string(*held);
-      }
-
       holder& operator=(const holder&) = delete;
-
 
     public: // representation
 
@@ -701,15 +643,6 @@ namespace vrpc {
 
     placeholder* content;
   };
-
-  inline std::ostream& operator<<(std::ostream& os, const Value& value) {
-    try {
-      if (value.content) os << value.content->format();
-    } catch (const std::exception& e) {
-      std::cerr << "Problem formatting value:" << e.what() << std::endl;
-    }
-    return os;
-  }
 
   inline bool operator==(const Value& lhs, const char& rhs) {
     return lhs.get<char>() == rhs;

